pkg/kubectl/cmd/config/list_context_test.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+)
+
+type listContextTest struct {
+	startingConfig clientcmdapi.Config
+	expectedOut    string
+}
+
+func TestListContextWithContext(t *testing.T) {
+	tconfig := clientcmdapi.Config{
+		Contexts: map[string]*clientcmdapi.Context{
+			"queen-anne-context": {AuthInfo: "black-user", Cluster: "pig-cluster", Namespace: "saw-ns"},
+			"shaker-context":     {AuthInfo: "green-user", Cluster: "horse-cluster", Namespace: "chisel-ns"}}}
+	test := listContextTest{
+		startingConfig: tconfig,
+		expectedOut:    "queen-anne-context\nshaker-context\n",
```

Comment:
It would also be useful to see the context properties (AuthInfo, Cluster, Namespace), in columnar form.